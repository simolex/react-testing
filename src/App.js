import logo from "./logo.svg";
import "./App.css";
import { useEffect, useState } from "react";

function App() {
  const [data, setData] = useState(null);
  useEffect(() => {
    setTimeout(() => {
      setData({});
    }, 100);
  }, []);
  return (
    <div className="App">
      {data && <div style={{ color: "red" }}>data</div>}
      <h1>Hello world!</h1>
      <button>Click me</button>
      <input type="text" placeholder="input value ..." />
    </div>
  );
}

export default App;
