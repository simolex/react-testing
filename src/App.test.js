import { render, screen } from "@testing-library/react";
import App from "./App";

describe("TEST App", () => {
  test("renders learn react link", () => {
    render(<App />);
    const helloWorldElem = screen.getByText(/hello world/i);
    const btnElem = screen.getByRole("button");
    const inputElem = screen.getByPlaceholderText(/input value/i);
    expect(helloWorldElem).toBeInTheDocument();
    expect(btnElem).toBeInTheDocument();
    expect(inputElem).toMatchSnapshot();
  });

  test("renders learn react link", async () => {
    render(<App />);
    // const helloWorldElem = screen.queryByText(/hello2/i);
    // expect(helloWorldElem).toBeNull();
    const helloWorldElem = await screen.findByText(/data/i);
    expect(helloWorldElem).toBeInTheDocument();
    expect(helloWorldElem).toHaveStyle({ color: "red" });
  });
});
