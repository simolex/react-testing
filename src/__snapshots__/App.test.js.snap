// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST App renders learn react link 1`] = `
<input
  placeholder="input value ..."
  type="text"
/>
`;

exports[`renders learn react link 1`] = `
<input
  placeholder="input value ..."
  type="text"
/>
`;
